### PR TITLE
Sonarqube integration

### PR DESCRIPTION
**Pull request by davidcauchi**

### CODE REVIEW

**Review by jmank88 — DISMISSED** [review]

File: .github/workflows/gauntlet.yml [hunk 1]
```diff
@@ -62,4 +62,12 @@ jobs:
           name: chainlink-cosmos
           authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
       - run: nix develop -c yarn install --frozen-lockfile
-      - run: nix develop -c yarn test:ci
+      - run: nix develop -c yarn test:ci --coverage
+
+      - name: Upload test coverage report
+        if: always()
```
Comment (cfal) [inline .github/workflows/gauntlet.yml:107]: nit: no need for `if` if it's always? same below

**Review by cfal — DISMISSED** [review]

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
           # working-directory: somedir
 
           # Optional: golangci-lint command line arguments.
-          args: --timeout=5m0s --tests=false
+          args: --timeout=5m0s --tests=false --out-format checkstyle:golangci-lint-report.xml
```
Comment (krebernisak) [inline .github/workflows/golangci-lint.yml:22]: Why do we keep the golangci-lint here but remove it from Solana in this [PR](https://github.com/smartcontractkit/chainlink-solana/pull/506/files)?

File: .github/workflows/relay.yml [hunk 3]
```diff
@@ -42,7 +42,17 @@ jobs:
         run: nix develop -c go build -v ./pkg/cosmos/...
 
       - name: Run unit tests
-        run: nix develop -c go test -v ./pkg/cosmos/...
+        run: nix develop -c go test -v -covermode=atomic ./pkg/cosmos/... -coverpkg=./... -coverprofile=coverage.txt
 
       - name: Run tests with the race detector enabled
-        run: nix develop -c go test -v ./pkg/cosmos/... -race -count=10
+        run: nix develop -c go test -v ./pkg/cosmos/... -race -count=10 -coverpkg=./... -coverprofile=race_coverage.txt
```
Comment (krebernisak) [inline .github/workflows/relay.yml]: Declaring custom CI steps inside a workflow is deprecated.

Please move these actions to Makefile and trigger them with make - we want all relevant project actions in a single place.

File: .github/workflows/golangci-lint.yml (hunk 2 above)
Comment (chudilka1) [reply .github/workflows/golangci-lint.yml:22]: Solana repo has the same config, please take a look [here](https://github.com/smartcontractkit/chainlink-solana/pull/506/files#diff-8c8835f2d3ac63273eb6b8a21e7f78bf1496eb82d29de54cae99b6282818699cR42)

File: .github/workflows/gauntlet.yml (hunk 1 above)
Comment (chudilka1) [reply .github/workflows/gauntlet.yml:107]: The previous step, if it fails, won't let this one get executed w/o `if always`; however, we still want to get the report no matter the results of the previous step.

**Review by chudilka1 — APPROVED** [review]

File: .github/workflows/sonar-scan.yml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # fetches all history for all tags and branches to provide more metadata for sonar reports
+
+      - name: Download Golangci report
+        uses: dawidd6/action-download-artifact@v2.27.0
+        with:
+          workflow: golangci-lint.yml
+          workflow_conclusion: ""
+          name_is_regexp: true
```
Comment (cfal) [inline .github/workflows/sonar-scan.yml]: name isn't a regexp and this can be false, same below: https://github.com/dawidd6/action-download-artifact#usage

**Review by cfal — DISMISSED** [review]
lgtm, left a nit about name_is_regexp

**Review by smickovskid — DISMISSED** [review]

**Review by smickovskid — DISMISSED** [review]

File: .github/workflows/sonar-scan.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+      - name: Download relay test reports
+        uses: dawidd6/action-download-artifact@v2.27.0
+        with:
+          workflow: relay.yml
+          workflow_conclusion: ""
+          name: go-test-results
+          if_no_artifact_found: warn
+      
+      - name: Download gauntlet test reports
```
Comment (chudilka1) [inline .github/workflows/sonar-scan.yml:59]: Subject to removal as the tests are aimed at Solana (contracts) not supported by Sonar.

File: .github/workflows/gauntlet.yml [hunk 6]
```diff
@@ -75,4 +75,12 @@ jobs:
         run: make contracts_compile
           
       - run: nix develop -c yarn install --frozen-lockfile
-      - run: nix develop -c yarn test:ci
+      - run: nix develop -c yarn test:ci --coverage
+
+      - name: Upload test coverage report
```
Comment (chudilka1) [inline .github/workflows/gauntlet.yml:106]: Upload artifact is a subject to removal () as the tests are aimed at Solana (contracts) not supported by Sonar.

File: .github/workflows/sonar-scan.yml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+          workflow_conclusion: ""
+          name: golangci-lint-report
+          if_no_artifact_found: warn
+
+      - name: Download relay test reports
+        uses: dawidd6/action-download-artifact@v2.27.0
+        with:
+          workflow: relay.yml
+          workflow_conclusion: ""
+          name: go-test-results
```
Comment (chudilka1) [inline .github/workflows/sonar-scan.yml:56]: These results are downloaded (see [here](https://github.com/smartcontractkit/chainlink-cosmos/actions/runs/6143363647/job/16667114504?pr=339#step:5:23)), but not uploaded see [here](https://github.com/smartcontractkit/chainlink-cosmos/actions/runs/6143363647/job/16667114504?pr=339#step:5:23). Most of the time it happens if the path to reports is found.

**Review by chudilka1 — CHANGES_REQUESTED** [review]

File: .github/workflows/gauntlet.yml (hunk 6 above)
Comment (smickovskid) [reply .github/workflows/gauntlet.yml:106]: These are mainly for gauntlet so we are good, the contracts tests here are copied over from the contracts dir which is excluded so they won't be scanned

File: .github/workflows/sonar-scan.yml (hunk 5 above)
Comment (smickovskid) [reply .github/workflows/sonar-scan.yml:59]: These are mainly for gauntlet so we are good, the contracts tests here are copied over from the contracts dir which is excluded so they won't be scanned

File: .eslintrc.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    "extends": [
+        "eslint:recommended",
+        "plugin:@typescript-eslint/recommended"
+    ],
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "ecmaVersion": "latest",
+        "sourceType": "module"
+    },
+    "ignorePatterns": ["./packages-ts/test/**", "**/node_modules/**"],
```
Comment (cfal) [inline .eslintrc.json]: there's no `packages-ts/test`, remove?

**Review by cfal — COMMENTED** [review]
lgtm overall - left a small comment. also could we rebase merge/develop? should fix the integration tests failure

**Review by smickovskid — APPROVED** [review]

**Pull request merged**